admin-console/src/main/webapp/src/pages/ApplicationPage.tsx

```diff
@@ -75,42 +92,53 @@ export const ApplicationPage: React.FunctionComponent = () => {
                         <Toolbar>
                             <ToolbarContent>
                                 <ToolbarItem>
-                                    <Button variant='primary' type='button' onClick={ toggle }> Create Event Type </Button>
+                                    <Button variant='primary' type='button'
+                                        onClick={ createEventType }> Create Event Type </Button>
                                     <Modal
                                         variant={ ModalVariant.medium }
                                         title={ `Create Event Type for ${ (applicationTypesQuery.loading ||
                                             applicationTypesQuery.payload?.status !== 200) ?
                                             <Spinner /> : applicationTypesQuery.payload.value.displayName }` }
-                                        isOpen={ isOpen }
-                                        onClose={ toggle }
+                                        isOpen={ showModal }
+                                        onClose={ () => setShowModal(false) }
```

Comment:
In general, is better to use `React.useCallback` (and `useMemo`) whenever is possible. It helps optimize the rendering (prevent unesesary renders)